flask_kafka_producer/__init__.py

```
@@ -0,0 +1,103 @@
+import json
+from uuid import UUID
+
+from pykafka import KafkaClient
+from pykafka.partitioners import random_partitioner, HashingPartitioner
+from flask import _app_ctx_stack as stack
+
+from .logging_middleware import LoggingMiddleware
+from .request_logger import RequestLogger
+
+__version__ = '0.0.1'
+
+
+class FlaskKafkaProducer(object):
+
+    def __init__(self, app, logging_topic, kafka_hosts=None):
+        self.init_app(app, logging_topic, kafka_hosts)
+
+    def init_app(self, app, logging_topic, kafka_hosts=None):
+        if not kafka_hosts:
+            kafka_hosts = app.config.get('KAFKA_HOSTS')
+        if not kafka_hosts:
+            raise Exception('Cannot initialize without kafka_hosts kwargs or KAFKA_HOSTS config value')
+        self.app = app
+        kafka_client = KafkaClient(hosts=kafka_hosts)
+        event_emitter = EventEmitter(kafka_client)
+        self.request_logger = RequestLogger(event_emitter)
+
+        app.wsgi_app = LoggingMiddleware(app, app.wsgi_app, self.request_logger, logging_topic)
+
+    def emit(self, event_type, event_subtype, payload, partition_key=None):
+        payload = {
+            'event_type': event_type,
+            'event_subtype': event_subtype,
+            'payload': payload
+        }
+        self.request_logger.emit(event_type, payload, partition_key)
+
+    def log(self, log_level, payload):
+        self.request_logger.log(log_level, payload)
+
+
+class EventEmitter(object):
+    def __init__(self, client):
+        self.client = client
+
+    def emit(self, topic, message, partition_key=None):
+        requires_partitioner = partition_key is not None
+        if isinstance(partition_key, str):
+            partition_key = partition_key.encode('utf-8')
+        elif isinstance(partition_key, UUID):
```

Maybe we should raise an exception if the user sends an unsupported var type.